File: app/src/main/java/com/sssprog/shoppingliststandalone/api/SimpleRxSubscriber.java
package com.sssprog.shoppingliststandalone.api;

import com.sssprog.shoppingliststandalone.utils.LogHelper;

import rx.Subscriber;

public class SimpleRxSubscriber<T> extends Subscriber<T> {
    @Override
    public void onCompleted() {

    }

    @Override
    public void onError(Throwable e) {
        LogHelper.printStackTrace(e);
    }

    @Override
    public void onNext(T t) {

    }
}

File: app/src/main/kotlin/com/sssprog/shoppingliststandalone/api/SimpleRxSubscriber.kt
package com.sssprog.shoppingliststandalone.api

import com.sssprog.shoppingliststandalone.utils.LogHelper
import rx.Subscriber

public open class SimpleRxSubscriber<T> : Subscriber<T>() {
    override fun onNext(result: T) {

    }

    override fun onError(e: Throwable?) {
        LogHelper.printStackTrace(e)
    }

    override fun onCompleted() {

    }

}
